### PR TITLE
[closes #4956] Extend netty-http test coverage

### LINKED ISSUE #4956: Improve netty-http test coverage
Improve `netty-http` consumer test coverage:

* Accessing `HttpRequest` / `HttpReponse` via `NettyHttpMessage`
* Basic authentication
* REST DSL
* Reverse proxy (E.g `from("netty-http:proxy")`)
* Specifying ACL on web resources with `SecurityConstraint` & `SecurityConstraintMapping`
* Wildcard routes (E.g `matchOnUriPrefix`)

### PR DESCRIPTION
**Pull request by avano**

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

**Comment by avano** [comment]
it seems that one test is flaky on windows, I'm looking at it

File: integration-tests/netty-http/src/test/java/org/apache/camel/quarkus/component/netty/http/NettyHttpJaasTestResource.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+ */
+package org.apache.camel.quarkus.component.netty.http;
+
+import java.util.Map;
+
+public class NettyHttpJaasTestResource extends NettyHttpTestResource {
+    @Override
+    public Map<String, String> start() {
+        // for JVM
+        System.setProperty("java.security.auth.login.config", "src/test/resources/config.jaas");
```
Comment (jamesnetherton) [inline integration-tests/netty-http/src/test/java/org/apache/camel/quarkus/component/netty/http/NettyHttpJaasTestResource.java:25]: Reading from the source tree will not work when the tests are run in the Quarkus Platform project.

Copy the file to `target` and use that as the path.

File: integration-tests/netty-http/src/test/java/org/apache/camel/quarkus/component/netty/http/NettyHttpJaasTestResource.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        final Map<String, String> properties = super.start();
+        // for native
+        properties.put("java.security.auth.login.config", "src/test/resources/config.jaas");
```
Comment (jamesnetherton) [inline integration-tests/netty-http/src/test/java/org/apache/camel/quarkus/component/netty/http/NettyHttpJaasTestResource.java:28]: As per previous comment about reading from the source tree.

**Comment by jamesnetherton** [comment]
By introducing a new `netty-http` itest module, we now have tests split between there and `integration-tests/http`. We should maybe consider moving the `netty-http` specific stuff our of `integration-tests/http` and into `integration-tests/netty-http`.

However, that could be non-trivial to do. So not sure if we should address it in this in this PR or do a follow-up. Maybe others have some thoughts on that.

There is also this proposal to add the HTTP tests to `integration-test-groups` https://github.com/apache/camel-quarkus/issues/3087.

**Comment by avano** [comment]
@jamesnetherton I can take a look at splitting the http tests in a separate PR

**Comment by jamesnetherton** [comment]
> @jamesnetherton I can take a look at splitting the http tests in a separate PR

Sounds good, thanks.

Do you mind rebasing with the latest work on main. Just want to make sure everything passes ok with the Quarkus 3.2.0 upgrade.

**Comment by avano** [comment]
> > @jamesnetherton I can take a look at splitting the http tests in a separate PR
> 
> Sounds good, thanks.
> 
> Do you mind rebasing with the latest work on main. Just want to make sure everything passes ok with the Quarkus 3.2.0 upgrade.

after the upgrade to 3.2.0 I needed to register more classes for reflection [here](https://github.com/avano/camel-quarkus/blob/issue-4956/extensions/netty-http/deployment/src/main/java/org/apache/camel/quarkus/component/netty/http/deployment/NettyHTTPProcessor.java#L38-L44)

**Comment by zbendhiba** [comment]
thanks @avano

**Pull request merged**